### PR TITLE
Add aria, key, and id props to Custom Button Injection

### PR DESCRIPTION
**Pull request by edwardlee-msft**
# What
Aria label, aria description, key, and id props added to custom buttons

Another change done:
Custom button ControlBarButton key attribute changed to uniqueId as key is not a prop locally known by ControlBarButton.

# Why
Support for accessibility and other features requiring unique ids.
https://skype.visualstudio.com/SPOOL/_workitems/edit/2958017
https://skype.visualstudio.com/SPOOL/_workitems/edit/2974780

# How Tested
Locally on MacOS on chrome browser
Example: 
Adding id's to custom buttons
image
image

*Note moreDrawer items component do not have props for aria labels.

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/communication-react/review/beta/communication-react.api.md [hunk 1]
```diff
@@ -1501,8 +1501,10 @@ export interface CustomCallWithChatControlButtonProps extends CustomControlButto
 
 // @beta
 export interface CustomControlButtonProps {
+    ariaDescription?: string;
+    ariaLabel?: string;
     disabled?: boolean;
-    // (undocumented)
+    key?: string | number;
```
Comment (JamesBurnside) [inline packages/communication-react/review/beta/communication-react.api.md:1507]: Is key sufficient if Contoso wants to provide a tooltip or flyout? From trying the [fluent callout](https://developer.microsoft.com/en-us/fluentui#/controls/web/callout) in the hackathon I needed to use `id`

Comment (JamesBurnside) [reply packages/communication-react/review/beta/communication-react.api.md:1507]: (note this could be a seperate bug I am thinking of)

Comment (edwardlee-msft) [reply packages/communication-react/review/beta/communication-react.api.md:1508]: Thanks for the comment James! 
Clarifying that id was an issue was definitely helpful. Updated to add id as well! :)

File: packages/react-composites/src/composites/CallWithChatComposite/CustomButton.tsx [hunk 2]
```diff
@@ -91,7 +94,7 @@ export const generateCustomCallWithChatDrawerButtons = (
         {allButtonProps
           .filter((buttonProps) => buttonProps.placement === key)
           .map((buttonProps, i) => ({
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallWithChatComposite/CustomButton.tsx:96]: can id be added here also for consistency?

File: packages/react-composites/src/composites/CallComposite/components/buttons/Custom.tsx [hunk 3]
```diff
@@ -79,7 +82,7 @@ export const generateCustomDrawerButtons = (
         {allButtonProps
           .filter((buttonProps) => buttonProps.placement === key)
           .map((buttonProps, i) => ({
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallComposite/components/buttons/Custom.tsx:84]: can id be added here also for consistency?

**Review by JamesBurnside — APPROVED** [review]

File: packages/react-composites/src/composites/CallComposite/components/buttons/Custom.tsx (hunk 3 above)
Comment (edwardlee-msft) [reply packages/react-composites/src/composites/CallComposite/components/buttons/Custom.tsx:84]: To add id to the drawer menu items I needed to make a small addition to the _DrawerMenuItemProps.

File: packages/react-components/review/beta/react-components.api.md [hunk 4]
```diff
@@ -505,6 +505,7 @@ export const _DrawerMenu: (props: _DrawerMenuProps) => JSX.Element;
 export interface _DrawerMenuItemProps {
     disabled?: boolean;
     iconProps?: IIconProps;
+    id?: string;
```
Comment (edwardlee-msft) [inline packages/react-components/review/beta/react-components.api.md:508]: @JamesBurnside , I had to add an id prop for _DrawerMenuItemProps to pass down an id to the drawer menu items.

Comment (JamesBurnside) [reply packages/react-components/review/beta/react-components.api.md:508]: Thanks for calling this out, looks perfect!

**Pull request merged**